packages/angular/src/generators/move-to-with-mf/move-to-with-mf.ts

```diff
@@ -0,0 +1,50 @@
+import type { Tree } from '@nrwl/devkit';
+import type { Schema } from './schema';
+
+import { readProjectConfiguration, formatFiles } from '@nrwl/devkit';
+import { getMfeProjects } from '../../utils/get-mfe-projects';
+import {
+  checkOutputNameMatchesProjectName,
+  checkSharedNpmPackagesMatchExpected,
+  getWebpackConfigPath,
+  isHostRemoteConfig,
+  parseASTOfWebpackConfig,
+  writeNewWebpackConfig,
+} from './lib';
+
+export default async function moveToWithMF(tree: Tree, schema: Schema) {
+  const projects = new Set(getMfeProjects(tree));
+
+  if (!projects.has(schema.project)) {
+    throw new Error(
+      `Could not find project \`${schema.project}\` in your workspace. Please check the name of the project you're wishing to convert exists.`
+    );
+  }
+
+  const project = readProjectConfiguration(tree, schema.project);
+  const pathToWebpackConfig = getWebpackConfigPath(project, schema.project);
+  const webpackAst = parseASTOfWebpackConfig(tree, pathToWebpackConfig);
+
+  if (!checkOutputNameMatchesProjectName(webpackAst, schema.project)) {
+    throw new Error(
+      `Cannot automatically migrate \`${schema.project}\` to \`withModuleFederation\` mirco frontend webpack config. 
+      \`uniqueName\` in webpack config (${pathToWebpackConfig}) does not match project name.`
+    );
+  }
+
+  if (!checkSharedNpmPackagesMatchExpected(webpackAst)) {
```

Comment:
Can't we still migrate them by using the `shared` property which allows customizing the defaults? I get it adds some more complexity to the generator, but I think it should be doable, isn't it?

Reply:
No because we'd have to create a bunch of if statements.

It would be _too_ complex.

our shared property that we expose is to override the default of sharing. So if they share the packages with a different config (which is what is being checked here) then we'd something like:

```js
(name, library) => {
 if(name === '@angular/core') {
   return configUserAlreadyHad.
 }
}
```

Reply:
Yeah, it will definitely add complexity. Still think is doable since we can easily collect the packages and specific properties that are changed with a minimal change to the existing `checkSharedNpmPackagesMatchExpected` and later we can just iterate over those to generate the `if` statements. I might be missing stuff, but it feels like we've done more complicated stuff than that in this same generator and others 🤷🏻 

That said, if we are not going to do that and still error, I think we should at least give the user a hint of what they can do to either avoid the error or to do the migrations themselves. If I look at the error message I wouldn't know what to do next.

Reply:
Good point. We should instruct them on what to do next. Which would be to consider if they need a custom configuration, and if not attempt to match the standard config we use.

I think the other side of this that I should give context on is that we're trying to enforce a single version policy. If one app has a custom config for `@angular/core` but other apps in the MFE architecture don't, then our migrator is already breaking the new policy we want to enforce. 

I think it's fine to allow people to opt-out with the escape hatch, but I don't think we should auto-generate that for them.

Reply:
> Because if we migrate them, then our generators generate conflicting package sharing configs, it's still "like" a failed migration, because stuff doesn't work after migrating.

The migration is not supposed to fix a broken setup, it's meant to convert from an old configuration to the new one. If the user has a broken setup, it will remain broken but use a different syntax. That's up to the user to have a working configuration. The generator wouldn't generate a wrong config, it generated the expected config given the existing configuration.

> the path our generators are taking doesn't match the path taken in the current MFE projects, and don't migrate them, I think that makes it clearer to the user that they need to change their setup if they want to follow our best practice path, or they will know that they need to use the escape hatch if they generate new remotes

It was already a best practice to use a single version policy, that's what we have been generating with the old version. Nothing has changed in that regard.  So, if a user has something different, it was a conscious decision to do it. At the very least it was definitely not something we generated for them and they *did* change it.

I'm fine with not generating it automatically if we want devs to think about the implications of breaking the single version policy. I think this is a valid reason to not do it automatically as long as we provide guidance on why we don't do it and what would be the next steps.

Reply:
Decision: It would be a good enchancement to have, but might still be out of scope for now. Will address in a future enchancement